hooks/notification/hook/Notifiers/TrelloNotifier.ts

```diff
@@ -0,0 +1,109 @@
+  public static readonly TRELLO_TOKEN = 'TRELLO_TOKEN';
+  public static readonly TRELLO_LIST = 'TRELLO_LIST';
+  public static readonly TRELLO_LABELS = 'TRELLO_LABELS';
+  public static readonly TRELLO_POS = 'TRELLO_POS';
+
+  protected type: NotifierType = NotifierType.TRELLO
+
+  constructor(channel: NotificationChannel, scan: Scan, findings: Finding[], args: Object) {
+    super(channel, scan, findings, args);
+  }
+
+  /**
+   * Trello cards API has a default endpoint https://api.trello.com/1/cards
+   * So if the TRELLO_CARDS_ENDPOINT env is not explicitly defined then default to that URL
+   */
+  public resolveEndPoint(): string {
+    if(TrelloNotifier.TRELLO_CARDS_ENDPOINT in process.env) { 
+      return super.resolveEndPoint(); 
+    }
+    return "https://api.trello.com/1/cards"
+  }
+
+  // The Trello hook will create a card for each finding
+  public async sendMessage(): Promise<void> {
+    for ( let idx in this.findings ) {
+      var jsonData = {
+          "key": this.getTrelloKey(),
+          "token": this.getTrelloToken(),
+          "idList": this.getTrelloList(),
+          "name": this.getCardName(this.findings[idx]),
+          "desc": this.getCardDescription(this.findings[idx]),
+          "pos": this.getTrelloPos()
+      }
+
+      // Add the labels only if defined    
+      if(this.getTrelloLabels().length > 0)
+        jsonData["idLabels"] = this.getTrelloLabels();
+
+      await this.sendJSONPostRequest(jsonData);
+    }
+  }
+
+  protected async sendJSONPostRequest( jsonData ) {
+    try {
+      await axios.post(this.resolveEndPoint(), jsonData)
+    } catch (e) {
+      console.log(`There was an Error sending the Message for the "${this.type}": "${this.channel.name}"`);
+      console.log(e);
+    }
+  }
+
+  private getTrelloKey(): string {
+    return process.env[TrelloNotifier.TRELLO_KEY];
+  }
+
+  private getTrelloToken(): string {
+    return process.env[TrelloNotifier.TRELLO_TOKEN];
+  }
+
+  private getTrelloList(): string {
+    return process.env[TrelloNotifier.TRELLO_LIST];
+  }
+
+  // If labels env not defined return empty string
+  private getTrelloLabels(): string {
+    if(TrelloNotifier.TRELLO_LABELS in process.env) { 
+      return process.env[TrelloNotifier.TRELLO_LABELS];
+    }
+    return ""
+  }
+
+  // If card pos env not defined return top
+  private getTrelloPos(): string {
+    if(TrelloNotifier.TRELLO_POS in process.env) { 
+      return process.env[TrelloNotifier.TRELLO_POS];
+    }
+    return "top"
+  }
+
+  // Constructs the card name from the finding
+  private getCardName(finding: Finding): string {
+    return "ZAP: " + finding.name + "(" + finding.severity + ")";
+  }
+
+  private getCardDescription(finding: Finding): string {
+    return  "Location: " + finding.location + "\n" +
+            "Description: " + finding.description + "\n" +
+            "Solution: " + finding.attributes["zap_solution"] + "\n" +
+            "Reference: " + finding.attributes["zap_reference"];
```

Comment:
Same as above for "zap". Running an nmap scan creates a card where Solution and Reference are undefined, which might be intended. But then it might be better to call the fields "ZAP - Solution" and "ZAP - Reference" and/or only include them if a ZAP scan was run. This could avoid some confusion and would create smaller cards without redundant information.

Reply:
Solution and Reference are now only added if the attributes map is not empty and the zap_solution and zap_solution exist in the map.